src/CommandExecutor.ts

```diff
@@ -63,6 +64,47 @@ class SystemFileExecutionStrategy extends CommandExecutionStrategy {
     }
 }
 
+class WindowsSystemFileExecutionStrategy extends CommandExecutionStrategy {
```

Comment:
Please create the hierarchy 
```
CommandExecutionStrategy
    SystemFileExecutionStrategy
        WindowsSystemFileExecutionStrategy
        UnixSystemFileExecutionStrategy
```

and extract the common logic to `SystemFileExecutionStrategy`.